tests/instruments/lecroy/test_lecroyT3DSO1204_with_device.py

```diff
@@ -0,0 +1,473 @@
+
+    This test suite, needs the following setup to work properly:
+        - A LeCroyT3DSO1204 device should be connected to the computer;
+        - The device's address must be set in the RESOURCE constant;
+        - A probe on Channel 1 must be connected to the Demo output of the oscilloscope.
+    """
+
+    ##################################################
+    # LeCroyT3DSO1204 device address goes here:
+    RESOURCE = "TCPIP::192.168.10.175::INSTR"
+    ##################################################
+
+    #########################
+    # PARAMETRIZATION CASES #
+    #########################
+
+    BOOLEANS = [False, True]
+    CHANNEL_COUPLINGS = ["AC 1M", "DC 1M", "GND"]
+    ACQUISITION_TYPES = ["normal", "average", "peak", "highres"]
+    TRIGGER_LEVELS = [0.125, 0.150, 0.175]
+    TRIGGER_SLOPES = ["negative", "positive", "window"]
+    ACQUISITION_AVERAGE = [4, 16, 32, 64, 128, 256]
+    WAVEFORM_POINTS = [100, 1000, 10000]
+    WAVEFORM_SOURCES = ["C1", "C2", "C3", "C4"]
+    CHANNELS = [1, 2, 3, 4]
+
+    SCOPE = LeCroyT3DSO1204(RESOURCE)
+
+    ############
+    # FIXTURES #
+    ############
+
+    @pytest.fixture
+    def make_scope(self):
+        return self.SCOPE
+
+    @pytest.fixture
+    def make_reseted_scope(self):
+        self.SCOPE.reset()
+        sleep(7)
+        return self.SCOPE
+
+    @pytest.fixture
+    def make_autoscale_scope(self):
+        self.SCOPE.reset()
+        sleep(7)
+        self.SCOPE.autoscale()
+        sleep(7)
+        return self.SCOPE
+
+    #########
+    # TESTS #
+    #########
+
+    # noinspection PyTypeChecker
+    def test_scope_connection(self):
+        bad_resource = "USB0::10893::45848::MY12345678::0::INSTR"
+        # The pure python VISA library (pyvisa-py) raises a ValueError while the
+        # PyVISA library raises a VisaIOError.
+        with pytest.raises((ValueError, VisaIOError)):
+            LeCroyT3DSO1204(bad_resource)
+
+    # Channel
+    def test_ch_current_configuration(self, make_autoscale_scope):
+        scope = make_autoscale_scope
+        scope.ch1.offset = 0
+        expected = {"channel": 1,
+                    "attenuation": 1.,
+                    "bandwidth_limit": False,
+                    "coupling": "DC 1M",
+                    "offset": 0.,
+                    "skew_factor": 0.,
+                    "display": True,
+                    "unit": "V",
+                    "volts_div": 0.05,
+                    "inverted": False,
+                    "trigger_coupling": "dc",
+                    "trigger_level": 0.150,
+                    "trigger_level2": 0.150,
+                    "trigger_slope": "positive"
+                    }
+        sleep(1)
+        actual = scope.ch(1).current_configuration
+        assert actual == expected
+
+    @pytest.mark.parametrize("ch_number", CHANNELS)
+    @pytest.mark.parametrize("case", BOOLEANS)
+    def test_ch_bwlimit(self, make_scope, ch_number, case):
+        scope = make_scope
```

Comment:
why these variable definitions? You can use the `make_scope` variable directly? Also elsewhere.

Reply:
I tried to mimick the unit test of Keysight DSOX1102G: tests/instruments/keysight/test_keysightDSOX1102G_with_device.py
I think the reason they do that is that it allows easy copy/past of unit tests. But if you want I can modify the unit tests and use make_scope directly.

Reply:
Yeah, use the variables directly, please.

Reply:
Done in commit 3839b540db95ffd57d9c5c5a6c20c007cdf07a72